src/qdsht.jl

```diff
@@ -0,0 +1,175 @@
+8-element Array{Float64,1}:
+  8.73574297951054e-9
+  4.166950654425497e-9
+  1.2135269740167775e-9
+  2.1570485709886677e-10
+  2.3441730449103528e-11
+  1.5335062911366657e-12
+  7.37934569592307e-14
+ -3.450933124215762e-15
+```
+"
+mul!(::Any, ::QDSHT, ::Any)
+
+"
+    ldiv!(Y, Q::QDSHT, A)
+
+Calculate the inverse quasi-discrete spherical Hankel transform of array `A` using the QDSHT
+`Q` and store the result in `Y`.
+
+# Examples
+```jldoctest
+julia> q = QDSHT(1e-2, 8); A = exp.(-q.r.^2/(1e-3*q.R)); Y = similar(A);
+julia> mul!(Y, q, A);
+julia> YY = similar(Y); ldiv!(YY, q, Y);
+julia> YY ≈ A
+true
+```
+"
+ldiv!(::Any, ::QDSHT, ::Any)
+
+"""
+    integrateR(A, Q::QDSHT; dim=1)
+
+Radial integral of `A`, over the aperture of `Q` in ``n+1``-dimensional real space.
+
+Assuming `A` contains samples of a function `f(r)` at sample points `Q.r`, then
+`integrateR(A, Q)` approximates ∫f(r)rⁿ dr from r=0 to r=∞.
+
+!!! note
+    `integrateR` and `integrateK` fulfill Parseval's theorem, i.e. for some array `A`,
+    `integrateR(abs2.(A), q)` and `integrateK(abs2.(q*A), q)` are equal, **but**
+    `integrateR(A, q)` and `integrateK(q*A, q)` are **not** equal.
+
+!!! warning
+    using `integrateR` to integrate a function (i.e. `A` rather than `abs2(A)`) is only
+    supported for the 0th-order QDSHT. For more details see [Derivations](@ref).
+
+# Examples
+```jldoctest
+julia> q = QDSHT(10, 128); A = exp.(-q.r.^2/2);
+julia> integrateR(abs2.(A), q) ≈ √π/4 # analytical solution of ∫exp(-r²)rⁿ dr from 0 to ∞
+true
+```
+"""
+integrateR(::Any, ::QDSHT)
+
+"""
+    integrateK(Ak, Q::QDSHT; dim=1)
+
+Radial integral of `A`, over the aperture of `Q` in ``n+1``-dimensional reciprocal space.
+
+Assuming `A` contains samples of a function `f(k)` at sample points `Q.k`, then
+`integrateR(A, Q)` approximates ∫f(k)kⁿ dk from k=0 to k=∞.
+
+!!! note
+    `integrateR` and `integrateK` fulfill Parseval's theorem, i.e. for some array `A`,
+    `integrateR(abs2.(A), q)` and `integrateK(abs2.(q*A), q)` are equal, **but**
+    `integrateR(A, q)` and `integrateK(q*A, q)` are **not** equal.
+
+# Examples
+```jldoctest
+julia> q = QDSHT(10, 128); A = exp.(-q.r.^2/2);
+julia> integrateR(abs2.(A), q) ≈ √π/4 # analytical solution of ∫exp(-r²)rⁿ dr from 0 to ∞
+true
+julia> Ak = q*A;
+julia> integrateK(abs2.(Ak), q) ≈ √π/4 # Same result
+true
+```
+"""
+integrateK(::Any, ::QDSHT)
+
+"""
+    onaxis(Ak, Q::QDSHT; dim=Q.dim)
+
+Calculate on-axis sample in space (i.e. at r=0) from transformed array `Ak`.
+
+# Examples
+```jldoctest
+julia> q = QDSHT(10, 128); A = exp.(-q.r.^2/2);
+julia> onaxis(q*A, q) ≈ 1 # should be exp(0) = 1
+true
+```
+"""
+function onaxis(Ak, Q::QDSHT; dim = Q.dim)
+    if Q.p != 0
+        throw(DomainError("on-axis samples can only be obtained for 0th-order transforms"))
+    end
+    j00ocn = inv(gamma((Q.n + 1) / 2) * 2^((Q.n - 1) / 2))
+    return j00ocn .* integrateK(Ak, Q; dim = dim)
+end
+
+function oversample(Q::QDSHT; factor::Int = 4)
```

Comment:
No, this is as it's intended. In #2, I added a second `oversample` method that just updates the plan as you see here. This is then called in a the larger `oversample` method typed on `AbstractQDHT` that acts on an array. This plan-specific method is necessary because the signatures of `QDHT` and `QDSHT` are different.

Reply:
Yes, I see now. Missed it in the sea of green.